### PR TITLE
UI: Add obs-frontend-api functions to create/delete profiles

### PR DESCRIPTION
**Pull request by SCG82**
### Description

Adds front-end API functions: 
+ `obs_frontend_create_profile(const char *name)`
+ `obs_frontend_duplicate_profile(const char *name)`
+ `obs_frontend_delete_profile(const char *profile)`

### How Has This Been Tested?

Tested on Win 10 & macOS 11.2:
* Create new profile with unique name - success
* Create new profile using an existing profile's name - no profile created, warning logged
* Create duplicate profile with unique name - success
* Create new profile using an existing profile's name - no profile created, warning logged
* Delete currently selected profile - success, current profile switches to a valid existing profile
* Delete profile not currently selected - success
* Delete profile that doesn't exist - warning logged
* Delete sole profile - silent fail

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/api-interface.cpp [hunk 1]
```diff
@@ -256,6 +256,20 @@ struct OBSStudioAPI : obs_frontend_callbacks {
...
+	{
+		QMetaObject::invokeMethod(main, "CreateProfile",
+					  WaitConnection(),
+					  Q_ARG(QString, name));
+	}
+
+	void obs_frontend_delete_profile(const char *profile) override
+	{
+		QMetaObject::invokeMethod(main, "DeleteProfile",
+					  WaitConnection(),
```
[the remark this answers is not in the document]
Comment (SCG82) [reply UI/api-interface.cpp]: Removed `WaitConnection()`

**Comment by RytoEX** [comment]
There's a lot of churn happening here.  Should this be set to draft until you're happy with it?

**Comment by SCG82** [comment]
Ready for review.

**Comment by WizardCM** [comment]
Commit prefix should simply be UI:

**Comment by SCG82** [comment]
Thoughts on nomenclature?
```
1. obs_frontend_create_profile  obs_frontend_delete_profile
2. obs_frontend_add_profile     obs_frontend_remove_profile
3. obs_frontend_new_profile     obs_frontend_remove_profile
4. obs_frontend_profile_create  obs_frontend_profile_delete
5. obs_frontend_profile_add     obs_frontend_profile_remove
```

**Comment by SCG82** [comment]
Should `obs_frontend_delete_profile` be renamed to `obs_frontend_remove_profile` to align with the Profile menu action?

**Comment by tt2468** [comment]
I would use the naming `obs_frontend_create_profile()` and `obs_frontend_remove_profile()`

**Comment by SCG82** [comment]
I didn't use `obs_frontend_create_profile` because there are two ways to create a profile: create a new profile, or duplicate the current profile.

**Comment by VodBox** [comment]
I would agree to use obs_frontend_create_profile for the adding of profiles. It matches with other parts of our APIs (like obs_source_create and obs_source_duplicate), and the internal functions are using the term create as well, so it makes it more consistent.

**Comment by SCG82** [comment]
Renamed `obs_frontend_new_profile` to `obs_frontend_create_profile`

**Pull request merged**